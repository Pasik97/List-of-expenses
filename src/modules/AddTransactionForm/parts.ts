import styled from 'styled-components';
import media from 'utils/media';

export const AddTransactionFormWrapper = styled.div`
   color: #1a1a1a;
   font-size: 18px;
   line-height: 21px;
   margin: 24px 0;
`;

export const Row = styled.div`
   margin-bottom: 8px;

   ${media.fromTablet} {
      display: grid;
      grid-template-columns: 3fr 1fr;
   }
`;

export const InputWrapper = styled.div`
   margin: 0 0 8px 0;

   ${media.fromTablet} {
      display: grid;
      grid-template-columns: 1fr 1fr;
   }
`;

export const InputLabel = styled.p`
   white-space: nowrap;
   margin: 0;

   ${media.onlyMobile} {
      margin-bottom: 6px;
   }

   ${media.fromTablet} {
      flex-grow: 1;
   }
`;

export const Input = styled.input`
   color: #1a1a1a;
   font-size: 18px;
   line-height: 21px;
   border: none;
   outline: none;
   border-bottom: 1px solid #dddddd;

   &:focus {
      border-bottom: 1px solid #1a1a1a;
   }

   ${media.fromTablet} {
      flex-grow: 1;
   }
`;

export const AddButton = styled.button`
   color: #1a1a1a;
   font-size: 18px;
   line-height: 21px;
   background: none;
   border: 2px solid #dddddd;
   padding: 6px 24px;

   &:disabled {
      color: #dddddd;
   }

   &:hover, &active {
      border-color: #1a1a1a;
      cursor: pointer;
   }

   ${media.fromTablet} {
      height: fit-content;
      width: fit-content;
      justify-self: flex-end;
   }
`;

export const ValidateInputWrapper = styled.div`
   display: flex;
   flex-direction: column;

   ${media.fromTablet} {
      margin-left: 16px;
      flex-grow: 1;
   }
`;

export const Validation = styled.p`
   margin: 8px 0 0 0;
   color: red;
   font-size: 13px;
   line-height: 16px;
   min-height: 16px;
`;
