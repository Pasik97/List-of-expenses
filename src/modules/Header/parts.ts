import styled from 'styled-components';

export const HeaderWrapper = styled.header`
   display: flex;
   align-items: baseline;
   justify-content: space-between;
   margin: 16px 0;
`;

export const Title = styled.h2`
   color: #1a1a1a;
   font-size: 28px;
   line-height: 32px;
   margin: 0 8px 0 0;
   font-weight: 700;
`;

export const ConversionRateWrapper = styled.div`
   display: flex;
   align-items: baseline;
   color: #1a1a1a;
   font-size: 16px;
   line-height: 21px;
`;

export const RateInput = styled.input`
   color: #1a1a1a;
   font-size: 16px;
   line-height: 21px;
   padding: 0;
   border: none;
   outline: none;
   border-bottom: 1px solid #dddddd;
   width: 60px;

   &:focus {
      border-bottom: 1px solid #1a1a1a;
   }
`;

export const ConversionLabel = styled.p`
   margin: 0;
   white-space: pre;
`;
