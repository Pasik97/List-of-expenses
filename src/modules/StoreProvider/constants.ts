import { ReactNode } from "react";
import Stores from "stores/Stores";

export interface StoreProviderProps {
   store: Stores;
   children: ReactNode;
}
