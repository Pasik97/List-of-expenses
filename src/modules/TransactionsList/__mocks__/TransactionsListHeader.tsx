const mock = () => '(TransactionsListHeader /)';

export default mock;
