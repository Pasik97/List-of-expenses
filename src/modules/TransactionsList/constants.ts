export enum SortType {
   Title = 'Title',
   AmountPln = 'AmountPln',
   AmountEur = 'AmountEur',
}

export enum SortDirection {
   Asc = 'Asc',
   Desc = 'Desc',
}

export interface TransactionListHeaderProps {
   currentSortType: SortType;
   currentSortDirection: SortDirection;
   onSortChange: (sortType: SortType, sortDirection: SortDirection) => void;
};
