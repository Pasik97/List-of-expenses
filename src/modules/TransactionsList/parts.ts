import styled from 'styled-components';
import media from 'utils/media';

export const TransactionsListWrapper = styled.div``;

export const Cell = styled.div`
   color: #1a1a1a;
   font-size: 16px;
   line-height: 21px;
   border-right: 1px solid #1a1a1a;
   padding: 6px;
   display: flex;
   align-items: center;

   &:last-of-type {
      border-right: none;
   }
`;

export const HeaderCell = styled(Cell)`
   font-weight: 700;
   background-color: #cccccc;
   
   &:hover, &:active {
      cursor: pointer;
   }
`;

export const Row = styled.div`
   display: grid;
   grid-template-columns: 1fr 1fr 1fr 1fr;
   border: 1px solid #1a1a1a;
   border-top: none;

   &:first-of-type {
      border-top: 1px solid #1a1a1a;
   }

   &:nth-of-type(2n+3) {
      ${Cell} {
         background-color: #eeeeee;
      }
   }

   ${media.fromTablet} {
      display: grid;
      grid-template-columns: 2fr 1fr 1fr 1fr;
   }

   ${media.fromDesktop} {
      display: grid;
      grid-template-columns: 3fr 1fr 1fr 1fr;
   }
`;

export const Label = styled.p`
   margin: 0;
   overflow-wrap: anywhere;
`;

export const DeleteButton = styled.button`
   color: #4d4d4d;
   font-size: 16px;
   line-height: 21px;
   border: none;
   outline: none;
   background: none;

   &:hover, &:active {
      cursor: pointer;
      color: #1a1a1a;
   }
`;
