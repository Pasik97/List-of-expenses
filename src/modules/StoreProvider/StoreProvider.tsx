import React, { createContext, } from 'react';
import { StoreProviderProps } from './constants';
import Stores from 'stores/Stores';

export const StoreProviderContext = createContext<Stores>({} as Stores);

const StoreProvider: React.FC<StoreProviderProps> = ({
   children,
   store,
}) => (
   <StoreProviderContext.Provider value={store}>
      {children}
   </StoreProviderContext.Provider>
);

export default StoreProvider;
