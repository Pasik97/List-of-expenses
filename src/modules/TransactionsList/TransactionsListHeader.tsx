import React from 'react';
import * as P from './parts';
import { SortDirection, TransactionListHeaderProps, SortType } from './constants';
import { observer } from 'mobx-react';

const TransactionsListHeader: React.FC<TransactionListHeaderProps> = observer(({
   currentSortType,
   currentSortDirection,
   onSortChange,
}) => {
   const isDescDirection = currentSortDirection === SortDirection.Desc;

   return (
      <P.Row>
         <P.HeaderCell
            onClick={() => onSortChange(SortType.Title, isDescDirection ? SortDirection.Asc : SortDirection.Desc)}
         >
            <P.Label>{currentSortType === SortType.Title && (isDescDirection ? 'ðŸ¡‡ ' : 'ðŸ¡… ')}Title</P.Label>
         </P.HeaderCell>
         <P.HeaderCell
            onClick={() => onSortChange(SortType.AmountPln, isDescDirection ? SortDirection.Asc : SortDirection.Desc)}
         >
            <P.Label>{currentSortType === SortType.AmountPln && (isDescDirection ? 'ðŸ¡‡ ' : 'ðŸ¡… ')}Amount (PLN)</P.Label>
         </P.HeaderCell>
         <P.HeaderCell
            onClick={() => onSortChange(SortType.AmountEur, isDescDirection ? SortDirection.Asc : SortDirection.Desc)}
         >
            <P.Label>{currentSortType === SortType.AmountEur && (isDescDirection ? 'ðŸ¡‡ ' : 'ðŸ¡… ')}Amount (EUR)</P.Label>
         </P.HeaderCell>
         <P.HeaderCell>
            <P.Label>Options</P.Label>
         </P.HeaderCell>
      </P.Row>
   )
});

export default TransactionsListHeader;
