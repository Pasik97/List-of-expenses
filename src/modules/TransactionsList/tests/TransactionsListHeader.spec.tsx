import { create } from "react-test-renderer";
import * as M from './mockData';
import TransactionsListHeader from "../TransactionsListHeader";
import { SortType, SortDirection } from "../constants";
import { HeaderCell } from "../parts";
import StoreProvider from "modules/StoreProvider/StoreProvider";

describe('<AddTransactionForm />', () => {
   describe(`with currentSortDirection set to ${SortDirection.Desc}`, () => {
      it(`should render table header cells with title with arrow down is sort type is ${SortType.Title}`, () => {
         const component = create(
            <StoreProvider store={M.mockStore}>
               <TransactionsListHeader
                  currentSortType={SortType.Title}
                  currentSortDirection={SortDirection.Desc}
                  onSortChange={jest.fn()}
               />
            </StoreProvider>,
         );

         expect(component.toJSON()).toMatchSnapshot();
      });

      it(`should render table header cells with amountPln with arrow down is sort type is ${SortType.AmountPln}`, () => {
         const component = create(
            <StoreProvider store={M.mockStore}>
               <TransactionsListHeader
                  currentSortType={SortType.AmountPln}
                  currentSortDirection={SortDirection.Desc}
                  onSortChange={jest.fn()}
               />
            </StoreProvider>,
         );

         expect(component.toJSON()).toMatchSnapshot();
      });

      it(`should render table header cells with amountEur with arrow down is sort type is ${SortType.AmountEur}`, () => {
         const component = create(
            <StoreProvider store={M.mockStore}>
               <TransactionsListHeader
                  currentSortType={SortType.AmountEur}
                  currentSortDirection={SortDirection.Desc}
                  onSortChange={jest.fn()}
               />
            </StoreProvider>,
         );

         expect(component.toJSON()).toMatchSnapshot();
      });

      it(`on title cell click should call onSortChange fn with ${SortType.Title} and ${SortDirection.Asc}`, () => {
         const mockOnSortChange = jest.fn()
         
         const component = create(
            <StoreProvider store={M.mockStore}>
               <TransactionsListHeader
                  currentSortType={SortType.AmountEur}
                  currentSortDirection={SortDirection.Desc}
                  onSortChange={mockOnSortChange}
               />
            </StoreProvider>,
         );
         
         component.root.findAllByType(HeaderCell)[0].props.onClick();

         expect(mockOnSortChange).toHaveBeenCalledWith(SortType.Title, SortDirection.Asc);
      });

      it(`on amountPln cell click should call onSortChange fn with ${SortType.AmountPln} and ${SortDirection.Asc}`, () => {
         const mockOnSortChange = jest.fn()
         
         const component = create(
            <StoreProvider store={M.mockStore}>
               <TransactionsListHeader
                  currentSortType={SortType.AmountEur}
                  currentSortDirection={SortDirection.Desc}
                  onSortChange={mockOnSortChange}
               />
            </StoreProvider>,
         );
         
         component.root.findAllByType(HeaderCell)[1].props.onClick();

         expect(mockOnSortChange).toHaveBeenCalledWith(SortType.AmountPln, SortDirection.Asc);
      });

      it(`on amountEur cell click should call onSortChange fn with ${SortType.AmountEur} and ${SortDirection.Asc}`, () => {
         const mockOnSortChange = jest.fn()
         
         const component = create(
            <StoreProvider store={M.mockStore}>
               <TransactionsListHeader
                  currentSortType={SortType.AmountEur}
                  currentSortDirection={SortDirection.Desc}
                  onSortChange={mockOnSortChange}
               />
            </StoreProvider>,
         );
         
         component.root.findAllByType(HeaderCell)[2].props.onClick();

         expect(mockOnSortChange).toHaveBeenCalledWith(SortType.AmountEur, SortDirection.Asc);
      });
   });

   describe(`with currentSortDirection set to ${SortDirection.Asc}`, () => {
      it(`should render table header cells with title with arrow up is sort type is ${SortType.Title}`, () => {
         const component = create(
            <StoreProvider store={M.mockStore}>
               <TransactionsListHeader
                  currentSortType={SortType.Title}
                  currentSortDirection={SortDirection.Asc}
                  onSortChange={jest.fn()}
               />
            </StoreProvider>,
         );

         expect(component.toJSON()).toMatchSnapshot();
      });

      it(`should render table header cells with amountPln with arrow up is sort type is ${SortType.AmountPln}`, () => {
         const component = create(
            <StoreProvider store={M.mockStore}>
               <TransactionsListHeader
                  currentSortType={SortType.AmountPln}
                  currentSortDirection={SortDirection.Asc}
                  onSortChange={jest.fn()}
               />
            </StoreProvider>,
         );

         expect(component.toJSON()).toMatchSnapshot();
      });

      it(`should render table header cells with amountEur with arrow up is sort type is ${SortType.AmountEur}`, () => {
         const component = create(
            <StoreProvider store={M.mockStore}>
               <TransactionsListHeader
                  currentSortType={SortType.AmountEur}
                  currentSortDirection={SortDirection.Asc}
                  onSortChange={jest.fn()}
               />
            </StoreProvider>,
         );

         expect(component.toJSON()).toMatchSnapshot();
      });

      it(`on title cell click should call onSortChange fn with ${SortType.Title} and ${SortDirection.Desc}`, () => {
         const mockOnSortChange = jest.fn()
         
         const component = create(
            <StoreProvider store={M.mockStore}>
               <TransactionsListHeader
                  currentSortType={SortType.AmountEur}
                  currentSortDirection={SortDirection.Asc}
                  onSortChange={mockOnSortChange}
               />
            </StoreProvider>,
         );
         
         component.root.findAllByType(HeaderCell)[0].props.onClick();

         expect(mockOnSortChange).toHaveBeenCalledWith(SortType.Title, SortDirection.Desc);
      });

      it(`on amountPln cell click should call onSortChange fn with ${SortType.AmountPln} and ${SortDirection.Desc}`, () => {
         const mockOnSortChange = jest.fn()
         
         const component = create(
            <StoreProvider store={M.mockStore}>
               <TransactionsListHeader
                  currentSortType={SortType.AmountEur}
                  currentSortDirection={SortDirection.Asc}
                  onSortChange={mockOnSortChange}
               />
            </StoreProvider>,
         );
         
         component.root.findAllByType(HeaderCell)[1].props.onClick();

         expect(mockOnSortChange).toHaveBeenCalledWith(SortType.AmountPln, SortDirection.Desc);
      });

      it(`on amountEur cell click should call onSortChange fn with ${SortType.AmountEur} and ${SortDirection.Desc}`, () => {
         const mockOnSortChange = jest.fn()
         
         const component = create(
            <StoreProvider store={M.mockStore}>
               <TransactionsListHeader
                  currentSortType={SortType.AmountEur}
                  currentSortDirection={SortDirection.Asc}
                  onSortChange={mockOnSortChange}
               />
            </StoreProvider>,
         );
         
         component.root.findAllByType(HeaderCell)[2].props.onClick();

         expect(mockOnSortChange).toHaveBeenCalledWith(SortType.AmountEur, SortDirection.Desc);
      });
   });
});
