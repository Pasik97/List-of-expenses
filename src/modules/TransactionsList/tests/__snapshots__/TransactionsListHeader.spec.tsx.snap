// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTransactionForm /> with currentSortDirection set to Asc should render table header cells with amountEur with arrow up is sort type is AmountEur 1`] = `
<div
  className="sc-hKwDye jGGgdQ"
>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Title
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Amount (PLN)
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      ðŸ¡… 
      Amount (EUR)
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Options
    </p>
  </div>
</div>
`;

exports[`<AddTransactionForm /> with currentSortDirection set to Asc should render table header cells with amountPln with arrow up is sort type is AmountPln 1`] = `
<div
  className="sc-hKwDye jGGgdQ"
>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Title
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      ðŸ¡… 
      Amount (PLN)
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Amount (EUR)
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Options
    </p>
  </div>
</div>
`;

exports[`<AddTransactionForm /> with currentSortDirection set to Asc should render table header cells with title with arrow up is sort type is Title 1`] = `
<div
  className="sc-hKwDye jGGgdQ"
>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      ðŸ¡… 
      Title
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Amount (PLN)
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Amount (EUR)
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Options
    </p>
  </div>
</div>
`;

exports[`<AddTransactionForm /> with currentSortDirection set to Desc should render table header cells with amountEur with arrow down is sort type is AmountEur 1`] = `
<div
  className="sc-hKwDye jGGgdQ"
>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Title
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Amount (PLN)
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      ðŸ¡‡ 
      Amount (EUR)
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Options
    </p>
  </div>
</div>
`;

exports[`<AddTransactionForm /> with currentSortDirection set to Desc should render table header cells with amountPln with arrow down is sort type is AmountPln 1`] = `
<div
  className="sc-hKwDye jGGgdQ"
>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Title
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      ðŸ¡‡ 
      Amount (PLN)
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Amount (EUR)
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Options
    </p>
  </div>
</div>
`;

exports[`<AddTransactionForm /> with currentSortDirection set to Desc should render table header cells with title with arrow down is sort type is Title 1`] = `
<div
  className="sc-hKwDye jGGgdQ"
>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      ðŸ¡‡ 
      Title
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Amount (PLN)
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
    onClick={[Function]}
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Amount (EUR)
    </p>
  </div>
  <div
    className="sc-gsDKAQ sc-dkPtRN yMCUW fZqwlO"
  >
    <p
      className="sc-eCImPb hZlHbZ"
    >
      Options
    </p>
  </div>
</div>
`;
